portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,6 +77,14 @@ private String convertToJson(ArrayList<String> messages) {
     return gson.toJson(messages);
   }
 
+    /**
+   * Converts an Arraylist<String> instance into a JSON string using gson
+   */
+  private String convertToJsonList(ArrayList<ArrayList<String>> messages) {
+    Gson gson = new Gson();
+    return gson.toJson(messages);
```

Comment:
This seems like not the best way to do it. I had hoped that I could use function overloading so I wouldn't have convertToJson AND convertToJsonList, but that errors, presumably because both types are ArrayLists(?).
I also considered using the generic < T > type but wasn't sure if that's good practice.

Reply:
Have you considered, returning a class object, instead of returning as list of strings. One advantage of returning typed info is you can make changes to the returning data structure while being backward compatible. This is more of a best practice.

see this example: https://www.baeldung.com/servlet-json-response

Reply:
Makes sense! Implemented the Comment class now – makes everything a lot cleaner :)